File: python-starter/src/benchmark/scenarios.py
"""Benchmark scenario definitions."""

from __future__ import annotations

from dataclasses import dataclass

from .types import SCENARIO_EXPECTATIONS, RequestContext


@dataclass(frozen=True)
class Scenario:
    name: str
    expected_terminal_state: str


BENCHMARK_SCENARIOS: list[Scenario] = [
    Scenario(name=name, expected_terminal_state=state.value)
    for name, state in SCENARIO_EXPECTATIONS.items()
]


def build_scenario_request_context(
    scenario: str, **overrides: object,
) -> RequestContext:
    kwargs: dict = {"scenario": scenario, **overrides}
    return RequestContext(**kwargs)

File: transfer/src/benchmark/scenarios.js
import { SCENARIO_EXPECTATIONS, createRequestContext } from "./types.js";

export const BENCHMARK_SCENARIOS = Object.freeze(
  Object.keys(SCENARIO_EXPECTATIONS).map((name) => ({
    name,
    expectedTerminalState: SCENARIO_EXPECTATIONS[name]
  }))
);

export function buildScenarioRequestContext(scenario, overrides = {}) {
  return createRequestContext({
    scenario,
    ...overrides
  });
}
